genanki/model.py

```diff
@@ -108,7 +109,7 @@ def to_json(self, now_ts, deck_id):
       "mod": now_ts,
       "name": self.name,
       "req": self._req,
-      "sortf": 0,
+      "sortf": self.sort_field_id,
```

Comment:
This is interesting, I didn't notice that the sort field index was encoded in the model's JSON representation. I wonder what happens when you try to set `sort_field` on a `Note` but leave `Model.sort_field_index` as 0. In that case, Anki will have conflicting information on what the sort field actually is.

My guess is that `sortf` in the model JSON (in `col.models`) is only used when creating new notes, and `notes.sfld` is used for sorting the data in the browser. So this inconsistency won't break Anki, but it could cause confusion if someone creates a new note based on a model from a genanki deck. This requires further investigation and it may be best to change the API to disable setting `Note.sort_field`.

Another funny thing I noticed is that when Anki generates a deck, the `notes.sfld` column is of `integer` type but contains text, which is [apparently a thing SQLite allows you to do](https://dba.stackexchange.com/questions/106364/text-string-stored-in-sqlite-integer-column). We do the same thing in `genanki`.

Reply:
Indeed this requires investigation. I can test this if you wish, but I don't have a lot of time these days...

Reply:
Yeah, mostly just noting this down so I can maybe investigate it later.